megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -2490,6 +2490,8 @@ public List<MovePath> getMovePathsAndSetNecessaryTargets(Entity mover, boolean f
                     // also return some paths that go a little slower than max speed
                     // in case the faster path would force an unwanted PSR or MASC check
                     prunedPaths.addAll(PathDecorator.decoratePath(prunedPath));
+                    // Return some of the already-computed unit paths as well.
+                    prunedPaths.addAll(getPrecognition().getPathEnumerator().getSimilarUnitPaths(mover.getId(), prunedPath));
```

Comment:
Is this method chaining any concern for null pointer exceptions?

Reply:
Shouldn't be; at this point in the function the backing objects all have to have been initialized.  If any of the collections are empty we just don't do any work.